flexibleengine/resource_flexibleengine_dis_stream_v2.go

```diff
@@ -0,0 +1,143 @@
+package flexibleengine
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/chnsz/golangsdk/openstack/dis/v2/streams"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var regexpStreamName = regexp.MustCompile(`^[a-zA-Z0-9_-]{1,64}$`)
+
+func resourceDisStreamV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDisStreamCreate,
+		Read:   resourceDisStreamRead,
+		Update: resourceDisStreamUpdate,
+		Delete: resourceDisStreamDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"stream_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(regexpStreamName,
+					"1 to 64 in length, only letters, digits, hyphens (-), and underscores (_) are allowed."),
+			},
+			"stream_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      streams.StreamTypeCommon,
+				ValidateFunc: validation.StringInSlice([]string{streams.StreamTypeCommon, streams.StreamTypeAdvanced}, false),
+			},
+			"partition_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+				// ForceNew: true, TODO: Check if update do the job
```

Comment:
seems that this parameter can not be updated in OTC

Reply:
done